pkg/controlplane/apiserver/aggregator.go

```diff
@@ -0,0 +1,69 @@
+func DefaultGenericAPIServicePriorities() map[schema.GroupVersion]APIServicePriority {
+	// The proper way to resolve this letting the aggregator know the desired group and version-within-group order of the underlying servers
+	// is to refactor the genericapiserver.DelegationTarget to include a list of priorities based on which APIs were installed.
+	// This requires the APIGroupInfo struct to evolve and include the concept of priorities and to avoid mistakes, the core storage map there needs to be updated.
+	// That ripples out every bit as far as you'd expect, so for 1.7 we'll include the list here instead of being built up during storage.
+	return map[schema.GroupVersion]APIServicePriority{
+		{Group: "", Version: "v1"}: {Group: 18000, Version: 1},
+		// to my knowledge, nothing below here collides
+		{Group: "events.k8s.io", Version: "v1"}:                      {Group: 17750, Version: 15},
+		{Group: "events.k8s.io", Version: "v1beta1"}:                 {Group: 17750, Version: 5},
+		{Group: "authentication.k8s.io", Version: "v1"}:              {Group: 17700, Version: 15},
+		{Group: "authentication.k8s.io", Version: "v1beta1"}:         {Group: 17700, Version: 9},
+		{Group: "authentication.k8s.io", Version: "v1alpha1"}:        {Group: 17700, Version: 1},
+		{Group: "authorization.k8s.io", Version: "v1"}:               {Group: 17600, Version: 15},
+		{Group: "certificates.k8s.io", Version: "v1"}:                {Group: 17300, Version: 15},
+		{Group: "certificates.k8s.io", Version: "v1alpha1"}:          {Group: 17300, Version: 1},
+		{Group: "rbac.authorization.k8s.io", Version: "v1"}:          {Group: 17000, Version: 15},
+		{Group: "apiextensions.k8s.io", Version: "v1"}:               {Group: 16700, Version: 15},
+		{Group: "admissionregistration.k8s.io", Version: "v1"}:       {Group: 16700, Version: 15},
+		{Group: "admissionregistration.k8s.io", Version: "v1beta1"}:  {Group: 16700, Version: 12},
+		{Group: "admissionregistration.k8s.io", Version: "v1alpha1"}: {Group: 16700, Version: 9},
+		{Group: "coordination.k8s.io", Version: "v1"}:                {Group: 16500, Version: 15},
+		{Group: "discovery.k8s.io", Version: "v1"}:                   {Group: 16200, Version: 15},
+		{Group: "discovery.k8s.io", Version: "v1beta1"}:              {Group: 16200, Version: 12},
+		{Group: "flowcontrol.apiserver.k8s.io", Version: "v1"}:       {Group: 16100, Version: 21},
+		{Group: "flowcontrol.apiserver.k8s.io", Version: "v1beta3"}:  {Group: 16100, Version: 18},
+		{Group: "flowcontrol.apiserver.k8s.io", Version: "v1beta2"}:  {Group: 16100, Version: 15},
+		{Group: "flowcontrol.apiserver.k8s.io", Version: "v1beta1"}:  {Group: 16100, Version: 12},
+		{Group: "flowcontrol.apiserver.k8s.io", Version: "v1alpha1"}: {Group: 16100, Version: 9},
+		{Group: "internal.apiserver.k8s.io", Version: "v1alpha1"}:    {Group: 16000, Version: 9},
+		{Group: "resource.k8s.io", Version: "v1alpha2"}:              {Group: 15900, Version: 9},
```

Comment:
Is this really generic? This seems to be related to Dynamic Resource Allocation, which is pod-specific if I'm not mistaken?

Reply:
Good question. I can see it has few types we might want to use in building some quota machinery:
https://pkg.go.dev/k8s.io/api@v0.30.0/resource/v1alpha2 
but in general I think you right.